employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/otp/OneTimePassword.java

```diff
@@ -0,0 +1,74 @@
+package org.optaweb.employeerostering.domain.otp;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Calendar;
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.validation.constraints.Email;
+
+import org.optaweb.employeerostering.exception.OtpInitRuntimeException;
+
+@Entity
+public class OneTimePassword {
+    private static final Integer TOKEN_LENGTH = 6;
+    private static final Integer TOKEN_MAX = 10; // exclusive
+    private static final Integer EXPIRY_MINUTES = 5;
+
+    private static SecureRandom SR = null;
+
+    static {
+        try {
+            SR = SecureRandom.getInstance("SHA1PRNG");
+        } catch (NoSuchAlgorithmException e) {
+            throw new OtpInitRuntimeException("Could not initialise OneTimePassword", e);
```

Comment:
You may want to consider throwing `IllegalStateException` instead. Also, if you prefer, consider...
```java
private static SecureRandom SR = makeSecureRandom("SHA1PRNG");

private static SecureRandom makeSecureRandom(String algorithm) {
    try {
        return SecureRandom.getInstance(algorithm);
    } catch (NoSuchAlgorithmException e) {
        throw new IllegalStateException("Could not initialise OneTimePassword", e);
    }
}
```